Makefile

```diff
@@ -0,0 +1,335 @@
+
+## Update conda environment using environment.yml
+conda-environment-update: conda-environment ta-lib
+	@-conda env update --name $(APP) -f environment.yml
+
+################################################################################
+#  Build App with Docker                                                       #
+################################################################################
+
+# App container name
+APP_CONTAINER := $(APP)
+
+# Strip /mnt if it exists, so Docker on WSL can find the path
+# TODO evaluate after WSL2 is released
+DOCKER_MOUNT_CONFIG_PATH := $(subst /mnt,,$(CONFIG_PATH))
+DOCKER_MOUNT_USER_DATA_PATH := $(subst /mnt,,$(PWD)/user_data/)
+
+# App mounts
+DOCKER_MOUNTS := \
+	--mount type=bind,source=$(DOCKER_MOUNT_CONFIG_PATH)/,target=/freqtrade/user_config \
+	--mount type=bind,source=$(DOCKER_MOUNT_USER_DATA_PATH),target=/freqtrade/user_data \
+	--mount type=bind,source=$(DOCKER_MOUNT_CONFIG_PATH)/tradesv3.sqlite,target=/freqtrade/tradesv3.sqlite \
+	--mount type=bind,source=$(DOCKER_MOUNT_CONFIG_PATH)/tradesv3.dryrun.sqlite,target=/freqtrade/tradesv3.dryrun.sqlite
+
+# App image filter
+APP_IMG_FILTER := docker ps -aq --filter ancestor=$(APP_CONTAINER) --format="{{.ID}}"
+
+## Build App container
+DOCKER_RUN_OSX := docker run -it --rm -e TZ=`ls -la /etc/localtime | cut -d/ -f8-9` 
+DOCKER_RUN_NIX := docker run -it --rm \
+	--mount type=bind,source=$(DOCKER_MOUNT_CONFIG_PATH)/localtime,target=/etc/localtime,readonly
+
+docker-app: build-docker-app
+build-docker-app: setup-create
+	@docker build -t $(APP_CONTAINER) -f $(APP_DOCKERFILE) .
+
+## Start app including START_ARGS
+docker-start-bot: docker-app
+ifeq ($(OSFLAG), OSX)
+	@$(DOCKER_RUN_OSX) -d --name $(APP_CONTAINER)-bot \
+		$(DOCKER_MOUNTS)\
+		$(APP_CONTAINER) $(START_ARGS)
+else 
+	$(DOCKER_RUN_NIX) -d --name $(APP_CONTAINER)-bot \
+		$(DOCKER_MOUNTS)\
+		$(APP_CONTAINER) $(START_ARGS)
+endif
+
+## open a shell in the container
+docker-shell:
+ifeq ($(OSFLAG), OSX)
+	@$(DOCKER_RUN_OSX) --name $(APP_CONTAINER)-shell -entrypoint=/bin/bash \
+		$(DOCKER_MOUNTS)\
+		$(APP_CONTAINER) -i
+else
+	@$(DOCKER_RUN_NIX) --name $(APP_CONTAINER)-shell --entrypoint=/bin/bash \
+		$(DOCKER_MOUNTS)\
+		$(APP_CONTAINER) -i
+endif
+
+# Run historical backtest
+TS := $(shell /bin/date "+%Y-%m-%d---%H-%M-%S")
+backtest:
+ifeq ($(OSFLAG), OSX)
+	@$(DOCKER_RUN_OSX) -d --name $(APP_CONTAINER)-backtest \
+		$(DOCKER_MOUNTS)\
+		$(APP_CONTAINER) $(START_ARGS) freqtrade backtesting --timerange=20180131- \
+		--refresh-pairs-cached  --export trades \
+		--export-filename=user_data/backtest_data/backtest_$(DEFAULT_STRATEGY)_$(TS).json
+else
+	@$(DOCKER_RUN_NIX) -d --name $(APP_CONTAINER)-backtest \
+		$(DOCKER_MOUNTS)\
+		$(APP_CONTAINER) $(START_ARGS) backtesting --timerange=20180131-\
+		--refresh-pairs-cached  --export trades \
+		--export-filename=user_data/backtest_data/backtest_$(DEFAULT_STRATEGY)_$(TS).json
+endif
+
+# Run backtest on live exchange
+backtest-live:
```

Comment:
`backtest --live` does not what most people think it does.
see this comment for explanation
https://github.com/freqtrade/freqtrade/issues/2052#issuecomment-514528038

i would remove this option since it's now deprecated (and honestly, never made much sense).